parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -91,7 +91,15 @@ if [[ ${ID} != "mariner" ]]; then
     logs_to_events "AKS.CSE.removeManDbAutoUpdateFlagFile" removeManDbAutoUpdateFlagFile
 fi
 
-if [[ "${GPU_NODE}" != "true" ]]; then
+export -f should_skip_nvidia_drivers
+skip_nvidia_driver_install=$(retrycmd_if_failure_no_stats 10 1 10 bash -c should_skip_nvidia_drivers)
```

Comment:
huh, haven't seen this pattern before, I guess this is the closest thing we have to "returning a bool"

probably better than trying to reuse exit codes and such

Reply:
yeah, this threw me for a loop...stdout vs exit codes and subshells is funky.

Reply:
feel like the only other (probably less) sane way to do it would be to use a global var and have it should_skip_nvidia_drivers just mutate it

Reply:
I might try moving the jq command separate from the curl retry, since that shouldn't fail, but I kind of like wrapping them together and that doesn't really solve the stdout/exit code issue...might just leave it.

Reply:
left this as-is for now, resolving unless you have thoughts